sphire/__init__.py

```diff
@@ -107,69 +107,76 @@ def getEnviron(cls):
-            correctCondaActivationCmd += ";"
-        return correctCondaActivationCmd
 
-    @classmethod
-    def defineBinaries(cls, env):
+        if cls._condaActivationCmd is None:
+            condaActivationCmd = os.environ.get(CONDA_ACTIVATION_CMD_VAR, "")
+            correctCondaActivationCmd = condaActivationCmd.replace(pw.Config.SCIPION_HOME + "/", "")
+            if not correctCondaActivationCmd:
+                print("WARNING!!: %s variable not defined. "
+                       "Relying on conda being in the PATH" % CONDA_ACTIVATION_CMD_VAR)
+            elif correctCondaActivationCmd[-1] != ";":
+                correctCondaActivationCmd += ";"
 
-        # try to get CONDA activation command
-        condaActivationCmd = cls.getCondaActivationCmd()
-        neededProgs = ['wget']
-        if not condaActivationCmd:
-            neededProgs.append('conda')
+            cls._condaActivationCmd = condaActivationCmd
 
-        cryolo_commands = cls.getInstallationCmd()
+        return cls._condaActivationCmd
 
-        envPath = os.environ.get('PATH', "")  # keep path since conda likely in there
-        installEnvVars = {'PATH': envPath} if envPath else None
-        env.addPackage('cryolo', version='1.4.0',
-                       tar='void.tgz',
-                       commands=cryolo_commands,
-                       neededProgs=neededProgs,
-                       default=False,
-                       vars=installEnvVars)
+    @classmethod
+    def defineBinaries(cls, env):
+
+        cls.addCryoloPackage(env, V1_5_3, default=bool(cls.getCondaActivationCmd()))
+        # Leave commented until released: cls.addCryoloPackage(env, V1_5_4_rc3)
 
         env.addPackage(CRYOLO_GENMOD, version=CRYOLO_GENMOD_20190516,
                        tar='void.tgz',
                        commands=[("wget ftp://ftp.gwdg.de/pub/misc/sphire/crYOLO-GENERAL-MODELS/" + CRYOLO_GENMOD_20190516_FN, CRYOLO_GENMOD_20190516_FN)],
                        neededProgs=["wget"],
-                       default=False)
-
+                       default=True)
 
     @classmethod
-    def getInstallationCmd(cls):
+    def getDependencies(cls):
+        # try to get CONDA activation command
+        condaActivationCmd = cls.getCondaActivationCmd()
+        neededProgs = ['wget']
+        if not condaActivationCmd:
+            neededProgs.append('conda')
+
+        return neededProgs
 
-        CRYOLO_INSTALLED = 'cryolo_installed'
+    @classmethod
+    def addCryoloPackage(cls, env, version, default=False):
 
+        CRYOLO_INSTALLED = 'cryolo_%s_installed' % version
+        ENV_NAME = getCryoloEnvName(version)
         # try to get CONDA activation command
         installationCmd = cls.getCondaActivationCmd()
 
         # Create the environment
-        installationCmd += 'conda create -y -n %s -c anaconda python=3.6 pyqt=5 cudnn=7.1.2 cython;' % DEFAULT_ENV_NAME
+        installationCmd += 'conda create -y -n %s -c anaconda python=3.6 ' \
```

Comment:
It will be safer to put a semicolon (;) before that command

Reply:
There was a semicolon before but that has a problem. In a command like:
`command1; command2 param;....`

if command1 fails the rest of the commands get executed and we were getting false succesfull installations when the first command was failing. And scipion believed installation was correct.

I've replaced the ; with && base on this:

https://stackoverflow.com/a/6152739/3157958

Reply:
Yes, I'm fine with &&, but in my case, it was adding 'conda' just after my activation command...and it didn't work at all.

Reply:
Then, to be consistent, also use '&&' here:
https://github.com/scipion-em/scipion-em-sphire/blob/update-to-1.5.4/sphire/__init__.py#L184